app/assets/javascripts/discourse/app/services/capabilities.js

```diff
@@ -11,12 +11,9 @@ class Capabilities {
   isIpadOS = ua.includes("Mac OS") && !/iPhone|iPod/.test(ua) && this.touch;
   isTabletScreen =
     this.touch &&
-    ((window.innerWidth >= 600 && window.innerWidth <= 1280) ||
-      (window.innerHeight >= 600 && window.innerHeight <= 1280));
-  isTablet =
-    this.isTabletScreen ||
-    this.isIpadOS ||
-    /iPad|Android(?!.*Mobile)|Tablet/.test(ua);
+    (window.innerWidth >= 600 || window.innerHeight >= 600) &&
+    window.matchMedia("(hover: none)").matches;
```

Comment:
Not sure whether to add the `window.matchMedia("(hover: none)").matches` check for tablets only or when adding `touch = ...` earlier in the file. Moving it to `touch` means that desktops with touchscreens wouldn't have the `touch` and `discourse-touch` classes in the `html` element.

Reply:
Oh, also, I removed the UA fallback for now because it looks redundant and can be misleading. The UAs on iPads don't add `iPad` (weird but true!)

![CleanShot 2025-01-15 at 10 28 42@2x](https://github.com/user-attachments/assets/5c1667cb-96b2-4e11-8ce8-d017e332b2dc)

`Tablet` isn't present either. We could keep `Android(?!.*Mobile)`, but I'm not sure we need it, the touch heuristic should match 100% of the time, right?